salt/modules/dockercompose.py

```diff
@@ -0,0 +1,438 @@
+	HAS_DOCKERCOMPOSE = True
+except ImportError:
+	HAS_DOCKERCOMPOSE = False
+
+MIN_DOCKERCOMPOSE = (1, 5, 0)
+VERSION_RE = r'([\d.]+)'
+
+log = logging.getLogger(__name__)
+debug = True
+
+__virtualname__ = 'dockercompose'
+dc_filename = "docker-compose.yml"
+
+
+def __virtual__():
+	if HAS_DOCKERCOMPOSE:
+		match = re.match(VERSION_RE, str(compose.__version__))
+		if match:
+			version = tuple([int(x) for x in match.group(1).split('.')])
+			if MIN_DOCKERCOMPOSE >= version:
+				return __virtualname__
+		else:
+			log.critical("Minimum version of docker-compose>=1.5.0")
+	return False
+
+
+def __standardize_result(status, message, data=None, debug_msg=None):
+	"""
+	Standardizes all responses
+
+	:param status:
+	:param message:
+	:param data:
+	:param debug_msg:
+	:return:
+	"""
+	result = {
+		'status': status,
+		'message': message
+	}
+
+	if data is not None:
+		result['return'] = data
+
+	if debug_msg is not None and debug:
+		result['debug'] = debug_msg
+
+	return result
+
+
+def __write_docker_compose(path, docker_compose):
+	"""
+	Write docker-compose to a temp directory
+	in order to use it with docker-compose ( config check )
+
+	:param path:
+
+	docker_compose
+		contains the docker-compose file
+
+	:return:
+	"""
+
+	if os.path.isdir(path) is False:
+		os.mkdir(path)
+	f = open(os.path.join(path, dc_filename), 'w')
+	if f:
+		f.write(docker_compose)
+		f.close()
+	else:
+		return __standardize_result(False, "Could not write docker-compose file in %s" % path, None, None)
+	project = __load_project(path)
+	if type(project) is dict:
+		os.remove(os.path.join(path, dc_filename))
+		return project
+	return path
+
+
+def __load_project(path):
+	"""
+	Load a docker-compose project from path
+
+	:param path:
+	:return:
+	"""
+	try:
+		project = get_project(path)
+	except Exception as inst:
+		return __handle_except(inst)
+	return project
+
+
+def __handle_except(inst):
+	"""
+	Handle exception and return a standart result
+
+	:param inst:
+	:return:
+	"""
+	return __standardize_result(False, "Docker-compose command %s failed" % (inspect.stack()[1][3]), "%s" % (inst), None)
+
+
+def _get_convergence_plans(project, service_names):
+	"""
+	Get action executed for each container
+
+	:param project:
+	:param service_names:
+	:return:
+	"""
+	ret = {}
+	plans = project._get_convergence_plans(project.get_services(service_names), ConvergenceStrategy.changed)
+	for cont in plans:
+		(action, container) = plans[cont]
+		if action == 'create':
+			ret[cont] = "Creating container"
+		elif action == 'recreate':
+			ret[cont] = "Re-creating container"
+		elif action == 'start':
+			ret[cont] = "Starting container"
+		elif action == 'noop':
+			ret[cont] = "Container is up to date"
+	return ret
+
+
+def create(path, docker_compose, **kwargs):
+	"""
+	Create and validate a docker-compose file into a directory
+
+	path
+		Path where the docker-compose file will be stored on the server
+
+	docker_compose
+		docker_compose file
+
+	:param kwargs:
+	:return:
```

Comment:
@jeanpralo, each api function here (functions not beginning with `_`) will need a CLI example at the end of the docstring:
```rst
    CLI Example:

    .. code-block:: bash

        salt '*' dockercompose.create ...
```